src/main/java/com/exasol/sql/expression/function/exasol/ExasolUdfFunction.java

```diff
@@ -0,0 +1,68 @@
+package com.exasol.sql.expression.function.exasol;
+
+import java.util.*;
+
+import com.exasol.sql.ColumnsDefinition;
+import com.exasol.sql.expression.ValueExpression;
+import com.exasol.sql.expression.function.AbstractFunction;
+
+/**
+ * This class represents a User Defined Function in the Exasol database.
+ */
+public class ExasolUdfFunction extends AbstractFunction {
+    private final Optional<ColumnsDefinition> emitsColumnsDefinition;
+
+    private ExasolUdfFunction(final String functionName, final Optional<ColumnsDefinition> emitsColumnsDefinition,
+            final List<ValueExpression> valueExpressions) {
+        super(functionName, valueExpressions);
+        this.emitsColumnsDefinition = emitsColumnsDefinition;
+    }
+
+    /**
+     * Create a new {@link ExasolUdfFunction} instance.
+     *
+     * @param functionName name of the function
+     * @param emitsColumnsDefinition column definitions for emits
+     * @param valueExpressions zero or more value expressions
+     * @return new {@link ExasolUdfFunction}
+     */
+    public static ExasolUdfFunction of(final String functionName, final ColumnsDefinition emitsColumnsDefinition,
+            final ValueExpression... valueExpressions) {
+        return new ExasolUdfFunction(functionName, Optional.of(emitsColumnsDefinition),
+                Arrays.asList(valueExpressions));
+    }
+
+    /**
+     * Create a new {@link ExasolUdfFunction} instance.
+     *
+     * @param functionName name of the function
+     * @param valueExpressions zero or more value expressions
+     * @return new {@link ExasolUdfFunction}
+     */
+    public static ExasolUdfFunction of(final String functionName, final ValueExpression... valueExpressions) {
+        return new ExasolUdfFunction(functionName, Optional.empty(), Arrays.asList(valueExpressions));
+    }
+
+    @Override
+    public boolean hasParenthesis() {
```

Comment:
I think `hasParameters` would be more expressive.

Reply:
Wait, sometimes function has no parameters, but has parenthesis. It's something different. I will rename it back...